File: app/src/main/java/fr/free/nrw/commons/media/model/ExtMetadata.java
package fr.free.nrw.commons.media.model;

import com.google.gson.annotations.SerializedName;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import fr.free.nrw.commons.utils.StringUtils;

public class ExtMetadata {
    @SuppressWarnings("unused") @SerializedName("DateTime") @Nullable
    private Values dateTime;
    @SuppressWarnings("unused") @SerializedName("ObjectName") @Nullable private Values objectName;
    @SuppressWarnings("unused") @SerializedName("CommonsMetadataExtension") @Nullable private Values commonsMetadataExtension;
    @SuppressWarnings("unused") @SerializedName("Categories") @Nullable private Values categories;
    @SuppressWarnings("unused") @SerializedName("Assessments") @Nullable private Values assessments;
    @SuppressWarnings("unused") @SerializedName("ImageDescription") @Nullable private Values imageDescription;
    @SuppressWarnings("unused") @SerializedName("DateTimeOriginal") @Nullable private Values dateTimeOriginal;
    @SuppressWarnings("unused") @SerializedName("Artist") @Nullable private Values artist;
    @SuppressWarnings("unused") @SerializedName("Credit") @Nullable private Values credit;
    @SuppressWarnings("unused") @SerializedName("Permission") @Nullable private Values permission;
    @SuppressWarnings("unused") @SerializedName("AuthorCount") @Nullable private Values authorCount;
    @SuppressWarnings("unused") @SerializedName("LicenseShortName") @Nullable private Values licenseShortName;
    @SuppressWarnings("unused") @SerializedName("UsageTerms") @Nullable private Values usageTerms;
    @SuppressWarnings("unused") @SerializedName("LicenseUrl") @Nullable private Values licenseUrl;
    @SuppressWarnings("unused") @SerializedName("AttributionRequired") @Nullable private Values attributionRequired;
    @SuppressWarnings("unused") @SerializedName("Copyrighted") @Nullable private Values copyrighted;
    @SuppressWarnings("unused") @SerializedName("Restrictions") @Nullable private Values restrictions;
    @SuppressWarnings("unused") @SerializedName("License") @Nullable private Values license;

    @NonNull public Values dateTime() {
        return dateTime != null ? dateTime : new Values();
    }

    @NonNull public Values dateTimeOriginal() {
        return dateTimeOriginal != null ? dateTimeOriginal : new Values();
    }

    @NonNull public Values licenseShortName() {
        return licenseShortName != null ? licenseShortName : new Values();
    }

    @NonNull public Values licenseUrl() {
        return licenseUrl != null ? licenseUrl : new Values();
    }

    @NonNull public Values license() {
        return license != null ? license : new Values();
    }

    @NonNull public Values imageDescription() {
        return imageDescription != null ? imageDescription : new Values();
    }

    @NonNull public Values objectName() {
        return objectName != null ? objectName : new Values();
    }

    @NonNull public Values usageTerms() {
        return usageTerms != null ? usageTerms : new Values();
    }

    @NonNull public Values artist() {
        return artist != null ? artist : new Values();
    }

    public class Values {
        @SuppressWarnings("unused,NullableProblems") @Nullable private String value;
        @SuppressWarnings("unused,NullableProblems") @Nullable private String source;
        @SuppressWarnings("unused,NullableProblems") @Nullable private String hidden;

        @NonNull public String value() {
            return StringUtils.defaultString(value);
        }

        @NonNull public String source() {
            return StringUtils.defaultString(source);
        }
    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/model/gallery/ExtMetadata.kt
package fr.free.nrw.commons.wikidata.model.gallery

import com.google.gson.annotations.SerializedName
import org.apache.commons.lang3.StringUtils

class ExtMetadata {
    @SerializedName("DateTime") private val dateTime: Values? = null
    @SerializedName("ObjectName") private val objectName: Values? = null
    @SerializedName("CommonsMetadataExtension") private val commonsMetadataExtension: Values? = null
    @SerializedName("Categories") private val categories: Values? = null
    @SerializedName("Assessments") private val assessments: Values? = null
    @SerializedName("GPSLatitude") private val gpsLatitude: Values? = null
    @SerializedName("GPSLongitude") private val gpsLongitude: Values? = null
    @SerializedName("ImageDescription") private val imageDescription: Values? = null
    @SerializedName("DateTimeOriginal") private val dateTimeOriginal: Values? = null
    @SerializedName("Artist") private val artist: Values? = null
    @SerializedName("Credit") private val credit: Values? = null
    @SerializedName("Permission") private val permission: Values? = null
    @SerializedName("AuthorCount") private val authorCount: Values? = null
    @SerializedName("LicenseShortName") private val licenseShortName: Values? = null
    @SerializedName("UsageTerms") private val usageTerms: Values? = null
    @SerializedName("LicenseUrl") private val licenseUrl: Values? = null
    @SerializedName("AttributionRequired") private val attributionRequired: Values? = null
    @SerializedName("Copyrighted") private val copyrighted: Values? = null
    @SerializedName("Restrictions") private val restrictions: Values? = null
    @SerializedName("License") private val license: Values? = null

    fun licenseShortName(): String = licenseShortName?.value ?: ""

    fun licenseUrl(): String = licenseUrl?.value ?: ""

    fun license(): String = license?.value ?: ""

    fun imageDescription(): String = imageDescription?.value ?: ""

    fun imageDescriptionSource(): String = imageDescription?.source ?: ""

    fun objectName(): String = objectName?.value ?: ""

    fun usageTerms(): String = usageTerms?.value ?: ""

    fun dateTimeOriginal(): String = dateTimeOriginal?.value ?: ""

    fun dateTime(): String = dateTime?.value ?: ""

    fun artist(): String = artist?.value ?: ""

    fun categories(): String = categories?.value ?: ""

    fun gpsLatitude(): String = gpsLatitude?.value ?: ""

    fun gpsLongitude(): String = gpsLongitude?.value ?: ""

    fun credit(): String = credit?.value ?: ""

    class Values {
        val value: String? = null
        val source: String? = null
        val hidden: String? = null
    }
}
